synapseclient/table.py

```diff
@@ -625,15 +625,13 @@ def _convert_df_date_cols_to_datetime(
     if diff_cols:
         raise ValueError("Please ensure that date columns are already in the dataframe")
     try:
-        df[date_columns] = (
-            df[date_columns].fillna(-1).astype(np.float64).astype(np.int64)
+        df[date_columns] = df[date_columns].applymap(
+            lambda x: np.int64(x) if pd.notnull(x) and str(x).isnumeric() else x
```

Comment:
This is a good solution. Couple things:

1. What if you provide it a number that is a float. Example: 1.1 <- That is technically a valid epoch time
2. I just noticed that you are doing try catch here, but I don't think its necessary here, since you're converting all the values that can be converted to integer and leaving the rest alone.

```
pd.to_datetime("test", unit="ms", utc=True)  # This will fail
```

Reply:
Keep the error message, but... maybe include valid epochtimes (integer, or doubles)